### PR TITLE
Brocade VDX switch integration as part of switch discovery

### PR DESCRIPTION
**Pull request by dpasupathi**
Changes related to on-http repo PR: Brocade VDX switch integration as part of switch discovery #331

Added SKU pack for Brocade-VDX switches

### CODE REVIEW

File: brocade-vdx/config.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+     "name": "Brocade VDX SKU",
+     "rules": [
+         {
+             "regex": "Network Operating System",
```
Comment (benbp) [inline brocade-vdx/config.json]: I'm worried this is a little too light of a definition. This may match plenty of other non-brocade switches as well.

**Comment by dpasupathi** [comment]
@benbp : This is true. But  as per my understanding "Network Operating System" is the OS name and i hope this will not be conflicting with other switch vendors. This is a first cut checkin, since other product lines are not yet released i will figure out an appropriate way to include both. Probably i can be more specific with some version number if you think it is apt:
    "regex": "Network Operating System Version: 7.1.0",

**Comment by anhou** [comment]
@dpasupathi  is there any other more reasonable characteristic for SKU? so that it could be differentiated from other SKUs,

**Comment by dpasupathi** [comment]
@anhou: Right now that is the best to match we can do based on our OS. Otherwise we need to add platform specific stuff. As mentioned we can add the firmware version number like 
"regex": "Network Operating System Version: 7.1.0",

**Comment by benbp** [comment]
@dpasupathi is there _any_ other data in the catalog that you can use to differentiate? Like a Brocade specific serial number, or at least a string containing the string "Brocade" ? Right now I believe it will match against Cisco switch catalogs as they also contain the string "Network Operating System"

**Comment by benbp** [comment]
Do you have an example of the version catalog JSON that we can see? (Would be good to add to the docs anyways when this goes in).

**Comment by dpasupathi** [comment]
@benbp : This is our "show version"  output.  only copyright has "Brocade "  keyword. 

Rackhdsw0# show version 
Network Operating System Software
Network Operating System Version: 7.1.0dhcpd
Copyright (c) 1995-2015 Brocade Communications Systems, Inc.
Firmware name:      7.1.0dhcpd
Build Time:         18:55:53 May 18, 2016
Install Time:       16:05:11 May 27, 2016
Kernel:             2.6.34.6
BootProm:           1.0.1
Control Processor:  e500mc with 4096 MB of memory

## Slot    Name    Primary/Secondary Versions                         Status

SW/0    NOS     7.1.0dhcpd                                         ACTIVE*
                        7.1.0dhcpd  
SW/1    NOS     7.1.0dhcpd                                         STANDBY
                       7.1.0dhcpd                                          

As part of cataloging the catalog script will send only one line as follows:

curl http://localhost:9090/api/2.0/nodes/57605cddce14875b7ad54bf1/catalogs | python -m json.tool
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   220  100   220    0     0  11328      0 --:--:-- --:--:-- --:--:-- 11578
[
    {
        "createdAt": "2016-06-14T19:37:04.536Z",
        "data": "Network Operating System Version: 7.1.0dhcpd",
        "id": "57605ce050565f587adae81b",
        "node": "57605cddce14875b7ad54bf1",
        "source": "version",
        "updatedAt": "2016-06-14T19:37:04.536Z"
    }
]

**Comment by benbp** [comment]
@dpasupathi Do you have a `switchshow` command on the CLI version you're using by any chance?

**Comment by yyscamper** [comment]
@dpasupathi : Could you please answer @benbp 's question? Since we still have concern on the sku rules.

File: brocade-vdx/workflows/brocade-vdx-config-deploy-graph.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                "friendlyName": "Deploy switch config",
+                "injectableName": "Task.Inline.Switch.Deploy.Brocade",
+                "implementsTask": "Task.Base.Linux.Commands",
+                "options": {
+                    "startupConfig": null,
+                    "startupConfigUri": "{{ api.base }}/templates/{{ options.startupConfig }}",
+                    "commands": [
+                        {
+                            "downloadUrl": "/api/1.1/templates/brocade-vdx-deploy-config.py"
+                        }
```
Comment (brianparry) [inline brocade-vdx/workflows/brocade-vdx-config-deploy-graph.json:22]: @dpasupathi The 1.1 API will be deprecated in the near future.  We should be using `/api/current/templates` instead.

**Comment by dpasupathi** [comment]
@benbp: Sorry i was on vacation and for few weeks. Yes i will modify the script so that it will catalog the following information "Hardware Rev                  : 137.4" from switchshow and i will add appropriate regex hope this will be fine?

**Comment by benbp** [comment]
@dpasupathi yes that seems more accurate, assuming no other vendors have output like "Hardware Rev : <N>".

**Comment by dpasupathi** [comment]
@benbp: Modified the regex specific to Brocade VDX and tested the same.

**Comment by benbp** [comment]
👍

**Review by benbp — APPROVED** [review]

**Review by brianparry — APPROVED** [review]
Looks good. :+1:

**Pull request merged**